### PR TITLE
Use one connection per data node for distributed copy insert

### PR DESCRIPTION
**Pull request by akuzm**
Currently we use one connection per chunk, and they all run the same
query `COPY destination_hypertable FROM STDIN`. Not sure what's the
point.

Part of https://github.com/timescale/timescaledb/pull/4285

### CODE REVIEW

File: tsl/src/remote/dist_copy.c [hunk 1]
```diff
@@ -54,13 +45,32 @@ typedef struct CopyDimensionInfo
...
-	List *cached_connections;
+	/*
+	 * Cached connections to data nodes.
+	 * Why do we need another layer of caching, when there is dist_txn layer
+	 * already? The API it provides is one function that "does everything
+	 * automatically", namely it's going to stop the COPY each time we request
+	 * the connection. This is not something we want to do for each row when
+	 * we're trying to do bulk copy.
+	 */
+	List *data_node_connections;
```
Comment (erimatnor) [inline tsl/src/remote/dist_copy.c:67]: We might not need this list if we instead use the standard connection cache (see comment below).

Comment (erimatnor) [reply tsl/src/remote/dist_copy.c:67]: It's ok to not refactor to use the connection cache in this PR, although I think this is what we should aim for as to not have multiple ways of dealing with connections.

Comment (akuzm) [reply tsl/src/remote/dist_copy.c:67]: The problem with using `remote_connection_cache` directly is that the remote chunk creation (`chunk_api_create_on_data_nodes`) uses the `dist_txn` layer.

**Comment by akuzm** [comment]
Merged as part of another PR.

**Pull request closed without merging**